src/fetch/FetchExample.re

```diff
@@ -0,0 +1,87 @@
+type dog = string;
+
+type arrayDogs = array(dog);
+
+type state = { dogs: arrayDogs, isLoading: bool, error: bool };
+
+type action =
+  | DogsFetch
+  | DogsFetched(arrayDogs)
+  | DogsFailedToFetch;
+
+module Decode = {
+  let dogs = json : arrayDogs =>
+  Json.Decode.(
+    json |> field("message", array(string))
+         |> Array.map(dog => dog)
+  );
+};
+
+let component = ReasonReact.reducerComponent("FetchExample");
+
+let make = _children => {
+  ...component,
+  initialState: () => { dogs: [||], isLoading: true, error: false },
```

Comment:
Putting `isLoading: true` here instead of setting it in `DogsFetch` still leaves them separated. What if you decide to `send(DogsFetch)` again after mounting for example? Use `UpdateWithSideEffects` to both set the state and trigger the side effect on the same action.